### PR TITLE
Jetpack REST API: Update endpoint docs

### PR DESCRIPTION
**Pull request by oskosk**
#### Changes proposed in this Pull Request:

Updates `docs/rest-api.md` to be up to date to latest endpoints behavior.

* Improves `/settings` endpoint documentation
* Clarifies that the modules endpoint are not super useful anymore.

#### Testing instructions:

* Check how the document in this branch looks like: [rest-api.md](https://github.com/Automattic/jetpack/blob/update/improve-api-docs-again/docs/rest-api.md)

#### Proposed changelog entry for your changes:

Improved documentation for Jetpack REST API endpoints.

### CODE REVIEW

File: docs/rest-api.md [hunk 1]
```diff
@@ -29,20 +109,75 @@ The root URL for the the API is found on the same page as:
...
-WP-API-compatible [capabilities document](http://v2.wp-api.org/guide/discovery/) for the endpoints registered by Jetpack.
+Core REST API-compatible [capabilities document](https://developer.wordpress.org/rest-api/using-the-rest-api/discovery/) for the endpoints registered by Jetpack.
 
 `GET /wp-json/jetpack/v4`
 
-## API Reference
+### Jetpack settings
 
-All endpoints return and accept JSON. Make sure you add the proper `content-type` header to your PUT/POST requests sending JSON objects.
+We call settings to any option that a module provides.
```
Comment (zinigor) [inline docs/rest-api.md]: Can you please rephrase this sentence? I don't understand what you meant to say here.

File: docs/rest-api.md [hunk 2]
```diff
@@ -83,19 +280,109 @@ Accepts a JSON object in the body like:
...
+
+
 ### Jetpack modules
 
-#### GET /wp-json/jetpack/v4/module
+#### GET /wp-json/jetpack/v4/module/all
 
 Get a list of all Jetpacks modules, its description, other properties and the module's options
 
+**Note**. The response has a big paylod in the body. Use it carefully.
```
Comment (zinigor) [inline docs/rest-api.md]: Typo in the word `payload`.

**Comment by oskosk** [comment]
Thanks @zinigor, I've updated this PR with two commits. Please, let me know if the Jetpack settings description is clearer now.

**Comment by zinigor** [comment]
Perfect, thank you!

**Pull request merged**